src/samVisualize/commands/visualizeSamTemplate/samVisualizationManager.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { getLogger, Logger } from '../../../shared/logger/logger'
+import * as vscode from 'vscode'
+import { SamVisualization } from './samVisualization'
+export class SamVisualizationManager {
+    private readonly extensionContext: vscode.ExtensionContext
+    // Maps a path to a template file to a SamVisualization
+    public readonly managedVisualizations: Map<string, SamVisualization> = new Map<string, SamVisualization>()
+
+    constructor(extensionContext: vscode.ExtensionContext) {
```

Comment:
Nit `public`, also `extensionContext` assignment in constructor

Reply:
What's wrong with `extensionContext` being assigned in the constructor?

Reply:
Nothing strictly wrong with it. If you declare the property `private readonly` in the constructor args then it automatically makes the assignment _and_ declaration, reducing the # of lines. Considering we don't have any lint rules set-up for this I wouldn't worry about it.